h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -158,12 +168,49 @@ private static int getDataRows(int[] chunks, Vec vec, int cols) {
      * @return A sum of chunk lengths. Possibly zero, if there are no chunks or the chunks are empty.
      */
     private static int sumChunksLength(int[] chunkIds, Vec vec) {
-        int totalChunkLength = 0;
-        for (int chunk : chunkIds) {
+      int totalChunkLength = 0;
+
+      for (int chunk : chunkIds) {
             totalChunkLength += vec.chunkLen(chunk);
         }
-        return totalChunkLength;
+      return totalChunkLength;
+    }
+
+  /**
+   * Counts zero-valued elements on a chunk
+   */
+  static class ChunkZeroCounter extends MrFun<ChunkZeroCounter> {
+      int _chunkLen = 0;
+      final int[] localChunkIDs;
+      final Chunk[] chunks;
+      public ChunkZeroCounter(final Vec weightsVector, final int[] localChunkIDs) {
+          this.localChunkIDs = localChunkIDs;
+
+          chunks = new Chunk[localChunkIDs.length];
+          for (int i = 0; i < localChunkIDs.length; i++) {
```

Comment:
I would move the chunk lookup to `map` - better parallelism

Reply:
Done. Moved to map.